developer/src/kmc-keyboard/src/compiler/key2.ts

```diff
@@ -60,7 +81,25 @@ export class Key2Compiler extends SectionCompiler {
   }
 
   public loadKeys(sections: GlobalSections, sect: Key2) {
-    for (let key of this.keyboard.keys.key) {
+    const usedKeys = this.allUsedKeyIds();
+
+    // TODO-LDML factor common code
+    // Need 'newer' (later) keys to override older ones.
+    const reverseKeys = [...this.keyboard.keys.key].reverse(); // newest to oldest
+    const alreadySeen = new Set<string>();
+    // filter out only the keys that haven't already been seen
+    const uniqueKeys = reverseKeys.filter(({id}) => {
+      if (!alreadySeen.has(id)) {
+        alreadySeen.add(id);
+        return true;
+      }
+      return false;
+    });
+
+    for (let key of uniqueKeys) {
+      if (!usedKeys.has(key.id)) {
+        continue; // unused key, skip
```

Comment:
In the future we will probably want to have linting here for unused keys -- excluding implied keys!

```suggestion
        // TODO-LDML: linting for unused keys
        continue; // unused key, skip
```